urbanairship-adapter/src/main/java/com/gimbal/urbanairship/GimbalAdapter.java

```diff
@@ -275,23 +351,26 @@ public void onResult(boolean enabled) {
         requestPermissionsTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, ACCESS_FINE_LOCATION);
     }
 
-    private void startAdapter() {
+
+    private void startAdapter(@NonNull String gimbalApiKey) {
         if (isAdapterStarted) {
             return;
         }
 
         preferences.edit()
-                .putBoolean(STARTED_PREFERENCE, true)
+                .putString(API_KEY_PREFERENCE, gimbalApiKey)
+                .putBoolean(STARTED_REFERENCE, true)
                 .apply();
 
         try {
+            Gimbal.setApiKey((Application) context.getApplicationContext(), gimbalApiKey);
```

Comment:
Where did Airship previously call `setApiKey`? Just wondering what the consequences of moving that call might be.

Reply:
In `restore()` and `enableApiKey()`.